### PR TITLE
User quota

### PR DESCRIPTION
**Pull request by AndyScherzinger**
revived PR by @przybylski, see also oC 102.

@przybylski is this PR ready for review in your eyes?

### CODE REVIEW

**Comment by przybylski** [comment]
yes

**Comment by AndyScherzinger** [comment]
Code wise :+1:

**Comment by AndyScherzinger** [comment]
cc @tobiasKaminsky

File: src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                // Parse the response
+                JSONObject respJSON = new JSONObject(response);
+                JSONObject respOCS = respJSON.getJSONObject(NODE_OCS);
+                JSONObject respData = respOCS.getJSONObject(NODE_DATA);
+                JSONObject quota    = respData.getJSONObject(NODE_QUOTA);
+                final Long quotaFree = quota.getLong(NODE_QUOTA_FREE);
+                final Long quotaUsed = quota.getLong(NODE_QUOTA_USED);
+                final Long quotaTotal = quota.getLong(NODE_QUOTA_TOTAL);
+                final Double quotaRelative = quota.getDouble(NODE_QUOTA_RELATIVE);
+
```
Comment (tobiasKaminsky) [inline src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java]: Remove empty line

File: src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+                // Result
+                result = new RemoteOperationResult(true, status, get.getResponseHeaders());
+                //Quota data in data collection
```
Comment (tobiasKaminsky) [inline src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java]: Space after //

File: src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+                ArrayList<Object> data = new ArrayList<Object>();
+                data.add(new Quota(quotaFree, quotaUsed, quotaTotal, quotaRelative));
+                result.setData(data);
+
```
Comment (tobiasKaminsky) [inline src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java]: remove empty line

File: src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                }
+            }
+        } catch (Exception e) {
+            result = new RemoteOperationResult(e);
+            Log_OC.e(TAG, "Exception while getting OC user information", e);
+
+        } finally {
+            get.releaseConnection();
+        }
+
```
Comment (tobiasKaminsky) [inline src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java]: empty line

File: src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        return result;
+    }
+
```
Comment (tobiasKaminsky) [inline src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java]: empty line

File: test_client/tests/src/com/owncloud/android/lib/test_project/test/GetUserQuotaTest.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    /* Files to download. These files must exist on the account */
+    private static final String IMAGE_PATH = "/fileToDownload.png";
+    private static final String IMAGE_PATH_WITH_SPECIAL_CHARS = "/@file@download.png";
+    private static final String IMAGE_NOT_FOUND = "/fileNotFound.png";
+
+    private String mFullPath2Image;
+    private String mFullPath2ImageWitSpecialChars;
+    private String mFullPath2ImageNotFound;
+    private String mDownloadedFilePath;
+    private TestActivity mActivity;
```
Comment (tobiasKaminsky) [inline test_client/tests/src/com/owncloud/android/lib/test_project/test/GetUserQuotaTest.java]: Remove all of not used finals / privates

**Comment by tobiasKaminsky** [comment]
Just a few minor code formattings.

File: src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java (hunk 1 above)
Comment (AndyScherzinger) [inline src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java]: fixed, see https://github.com/nextcloud/android-library/pull/10/commits/9c2391365252737d8ddd5bf14accfa0f9b9af670

File: src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java (hunk 2 above)
Comment (AndyScherzinger) [inline src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java]: fixed, see https://github.com/nextcloud/android-library/pull/10/commits/9c2391365252737d8ddd5bf14accfa0f9b9af670

File: src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java (hunk 3 above)
Comment (AndyScherzinger) [inline src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java]: fixed, see https://github.com/nextcloud/android-library/pull/10/commits/9c2391365252737d8ddd5bf14accfa0f9b9af670

File: src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java (hunk 4 above)
Comment (AndyScherzinger) [inline src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java]: fixed, see https://github.com/nextcloud/android-library/pull/10/commits/9c2391365252737d8ddd5bf14accfa0f9b9af670

File: src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java (hunk 5 above)
Comment (AndyScherzinger) [inline src/com/owncloud/android/lib/resources/users/RemoteGetUserQuotaOperation.java]: fixed, see https://github.com/nextcloud/android-library/pull/10/commits/9c2391365252737d8ddd5bf14accfa0f9b9af670

File: test_client/tests/src/com/owncloud/android/lib/test_project/test/GetUserQuotaTest.java (hunk 6 above)
Comment (AndyScherzinger) [inline test_client/tests/src/com/owncloud/android/lib/test_project/test/GetUserQuotaTest.java]: fixed, see https://github.com/nextcloud/android-library/pull/10/commits/9c2391365252737d8ddd5bf14accfa0f9b9af670

**Pull request merged**